lib/pos_blockchain.go

```diff
@@ -337,15 +333,29 @@ func (bc *Blockchain) validateTimeoutQC(desoBlock *MsgDeSoBlock, validatorSet []
 // validateAncestorsExist checks that all ancestors of this block exist in the block index.
 // If an ancestor is not found, we'll return the block hash of the missing ancestor so the
 // caller can request this block.
-func (bc *Blockchain) validateAncestorsExist(desoBlock *MsgDeSoBlock) (_missingBlockHash *BlockHash, _err error) {
+func (bc *Blockchain) validateAncestorsExist(parentHash BlockHash) *BlockHash {
 	// Notes: starting from the block passed in, we'll look for the parent in the block index.
 	// 1. If the parent does not appear in the block index, we'll return the parent's hash.
 	// 2. If the parent exists in the block index AND is in the best chain, we can safely assume
 	//    that all ancestors exist in the block index.
 	// 3. If the parent exists in the block index but is not in the best chain, we repeat from
 	//    step 1 with the parent as the block passed in.
-	// TODO: Implement me
-	return nil, errors.New("IMPLEMENT ME")
+
```

Comment:
Do we need to check orphans?

Reply:
I don't think so. Just checking if the block's parent exists in the block index is sufficient. But you can always iterate if needed